packages/framework/src/Component/Grid/QueryBuilderDataSource.php

```diff
@@ -18,12 +17,12 @@ class QueryBuilderDataSource implements DataSourceInterface
     /**
      * @param \Doctrine\ORM\QueryBuilder $queryBuilder
      * @param string $rowIdSourceColumnName
-     * @param string|null $hint
+     * @param array $hints
      */
     public function __construct(
         protected readonly QueryBuilder $queryBuilder,
         protected readonly string $rowIdSourceColumnName,
-        protected readonly ?string $hint = SortableNullsWalker::class,
+        protected array $hints = [],
```

Comment:
Doesn't this change the default value? 

If someone uses the query builder data source, will it still have the sortablenullswalker?
Maybe when we now have factories, we could make this option mandatory (remove the default value)?